src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h

```diff
@@ -618,72 +669,75 @@ namespace sirf {
+		std::vector<KSpaceSubset> get_kspace_sorting() const { return this->sorting_; }
 
-		virtual std::vector<int> get_flagged_acquisitions_index(const std::vector<ISMRMRD::ISMRMRD_AcquisitionFlags> flags) const;
-		virtual std::vector<int> get_slice_encoding_index(const unsigned kspace_encode_step_2) const;
+		//! Function to go through Acquisitions and assign them to their k-space dimension
+		/*!
+		 * All acquisitions belong to only one subset in a multi-dimensional k-space. This function goes through
+		 * all acquisitions in the container, extracts their subset (i.e. which slice contrast etc.) and stores
+		 * this information s.t. consisten subsets (i.e. all acquisitions belonging to the same slice) can be
+		 * extracted.
+		 */
+		void organise_kspace();
 
+		virtual std::vector<int> get_flagged_acquisitions_index
+			(const std::vector<ISMRMRD::ISMRMRD_AcquisitionFlags> flags) const;
+		virtual std::vector<int> get_slice_encoding_index
+			(const unsigned kspace_encode_step_2) const;
 
-        virtual void get_subset(MRAcquisitionData& subset, const std::vector<int> subset_idx) const;
-        virtual void set_subset(const MRAcquisitionData &subset, const std::vector<int> subset_idx);
+		virtual void get_subset(MRAcquisitionData& subset, const std::vector<int> subset_idx) const;
+		virtual void set_subset(const MRAcquisitionData &subset, const std::vector<int> subset_idx);
 
 		std::vector<int> index() { return index_; }
 		const std::vector<int>& index() const { return index_; }
 
 		int index(int i) const
 		{
 			const std::size_t ni = index_.size();
-			if (i < 0 || (ni > 0 && static_cast<std::size_t>(i) >= ni) || static_cast<unsigned>(i) >= number())
+			if (i < 0 || (ni > 0 && static_cast<std::size_t>(i) >= ni) 
+				|| static_cast<unsigned>(i) >= number())
 				THROW("Aquisition number is out of range");
 			if (ni > 0)
 				return index_[i];
 			else
 				return i;
 		}
 
-    	/*! 
-    		\brief Reader for ISMRMRD::Acquisition from ISMRMRD file. 
-      		*	filename_ismrmrd_with_ext:	filename of ISMRMRD rawdata file with .h5 extension.
-      		* 
-      		* In case the ISMRMRD::Dataset constructor throws an std::runtime_error the reader catches it, 
-      		* displays the message and throws it again.
-			* To avoid reading noise samples and other calibration data, the TO_BE_IGNORED macro is employed
-			* to exclude potentially incompatible input. 
-    	*/
+		/*!
+		\brief Reader for ISMRMRD::Acquisition from ISMRMRD file.
+		* filename_ismrmrd_with_ext: filename of ISMRMRD rawdata file with .h5 extension.
+		* In case the ISMRMRD::Dataset constructor throws an std::runtime_error the reader catches it,
+		* displays the message and throws it again.
+		* To avoid reading noise samples and other calibration data, IgnoreMask may be employed
+		* to exclude potentially incompatible input.
+		*/
 		void read(const std::string& filename_ismrmrd_with_ext, int all = 0);
```

Comment:
what is the `all` parameter? not documented

Reply:
documented